mc-java/src/main/kotlin/io/spine/tools/mc/java/gradle/Artifacts.kt

```diff
@@ -114,71 +83,131 @@ internal val toolBase: Artifact by lazy {
+            version = this@McJava.version
+            extension = JAR_EXTENSION
+        }
+    }
 
-private val validationJavaBundleDependency =
-    ThirdPartyDependency(VALIDATION_GROUP, "spine-validation-java-bundle")
+    /**
+     * The Maven artifact containing the `spine-mc-java-rejection` module.
+     */
+    @JvmStatic
+    @get:JvmName("rejection")
+    internal val rejection: Artifact by lazy {
+        artifact {
+            useSpineToolsGroup()
+            setName("spine-mc-java-rejection")
+            setVersion(this@McJava.version)
+            setExtension(JAR_EXTENSION)
+        }
+    }
 
-private val validationJavaRuntimeDependency =
-    ThirdPartyDependency(VALIDATION_GROUP, "spine-validation-java-runtime")
+    /**
+     * The Maven artifact containing the `spine-mc-java-annotation` module.
+     */
+    @get:JvmName("annotation")
+    @JvmStatic
+    internal val annotation: Artifact by lazy {
+        artifact {
+            useSpineToolsGroup()
+            setName("spine-mc-java-annotation")
+            setVersion(this@McJava.version)
+            setExtension(JAR_EXTENSION)
+        }
+    }
 
-private val validationVersion: String by lazy {
-    versions.versionOf(validationJavaDependency).orElseThrow()
+    /**
+     * The Maven artifact containing the `spine-mc-java-plugins:all` fat JAR artifact.
+     */
+    @JvmStatic
+    @get:JvmName("allPlugins")
+    internal val allPlugins: Artifact by lazy {
+        artifact {
+            useSpineToolsGroup()
+            setName(SPINE_MC_JAVA_ALL_PLUGINS_NAME)
+            setVersion(this@McJava.version)
+            setClassifier(ALL_CLASSIFIER)
+            setExtension(JAR_EXTENSION)
+        }
+    }
 }
 
 /**
- * The Maven artifact containing the `spine-validation-java-extensions` module.
+ * Artifacts of the Spine Validation library.
  */
-@get:JvmName("validationJavaBundle")
-internal val validationJavaBundle: Artifact by lazy {
-    artifact {
-        dependency = validationJavaBundleDependency
-        version = validationVersion
+internal object Validation {
+
+    @Suppress("ConstPropertyName")
+    private const val group = "io.spine.validation"
+    private val javaCodegen = MavenDependency(group, "spine-validation-java")
+    private val javaCodegenBundle = MavenDependency(group, "spine-validation-java-bundle")
+    private val javaRuntime = MavenDependency(group, "spine-validation-java-runtime")
+
+    private fun validationVersion(version: String = ""): String =
+        version.ifEmpty {
+            versions.versionOf(javaCodegen).orElseThrow {
+                error("Unable to load the version of `$javaCodegen`.")
+            }
+        }
+
+    /**
+     * The Maven artifact containing the `spine-validation-java-bundle` module.
+     *
+     * @param version
+     *         the version of Validation library to be used.
+     *         If empty, the version of the dependency used at the build time is used.
```

Comment:
```suggestion
     *         If empty, the version of the build time dependency is used.
```

I would just try to get rid of two `used` in the same sentence.